test-data/unit/fine-grained.test

```diff
@@ -8673,3 +8673,37 @@ def f(x: Union[int, str]) -> None: ...
 [out]
 ==
 ==
+
+[case testReprocessModuleTopLevelWhileMethodDefinesAttrExplicit]
+import a
+[file a.py]
+from b import B
+
+B().x
+
+[file a.py.3]
+from b import B
+
+x = B().x
```

Comment:
It's not clear to me why we have this update. Maybe add comment? Also, it might be easier to read this if the updates were in order (.2 comes before .3 and all initial versions come first).